### PR TITLE
plan: fix the join plan's hints.

### PR DESCRIPTION
**Pull request by winoros**
Hints for join will fail when construct semi join plan.
PTAL @hanfei1991 @zz-jason @lamxTyler

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -1768,6 +1770,13 @@ func (b *planBuilder) buildSemiJoin(outerPlan, innerPlan LogicalPlan, onConditio
 			joinPlan.JoinType = SemiJoin
 		}
 	}
+	if b.TableHints() != nil {
```
Comment (lamxTyler) [inline plan/logical_plan_builder.go]: Could we put the declaration of `outerAlias` and `innerAlias` here?

Comment (winoros) [reply plan/logical_plan_builder.go]: Agree.

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**